### DISCUSSION TITLE
Starlette TestClient Does Not Redirect To External URL When Follow Redirects Is Set To True

### DISCUSSION THREAD

**Discussion by infohash** [encode/starlette#1911] [Potential Issue] [answered]
In my test method, I am testing a request to an endpoint that always returns a redirect response. The URL in the redirect response is an external URL (for external API request) to which the TestClient instance method should make a follow up request if the `follow_redirects` is set to True. Because it is a test case, I have mocked the response of the external API request when the follow up redirect will happen.

```python
with TestClient(app=app, base_url="https://client.example.com") as client:
    client.get("/endpoint", follow_redirects=True)
```

If `follow_redirects` is `False`, I can see the redirect response but when it's `True`, it returns 404 because it cannot find the resource.

The problem does not occur when redirect happens to any endpoint under the same `base_url`.

# Expected Behaviour
It should make a follow up request to the redirected external URL whether or not the resource is available and let the user to take care of patching the response.

Is this the desired behaviour already or am I doing it wrong? I am using Starlite framework that uses Starlette TestClient to mock requests.

**Comment by Kludex** [comment, marked as answer]
I believe this is outdated, so I'll close it. Let me know if it's not.
